### ISSUE TITLE
Type Hinting in Python

### ISSUE THREAD

**Issue by warmachine028** [girlscript/winter-of-contributing#4300]
### Description

A detailed  explanation of Type Hinting:
- What is Type Hinting?
- How we use type Hinting in python?
- Why we need type hinting in a Dynamic Type Language?
- Basics
- Advanced 

### Domain

Python_Domain

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

**Comment by adityabisoi** [comment]
Go ahead

**Comment by warmachine028** [comment]
> Go ahead

Hey I have Completed the documentation. Can you have a look ? #4422

**Comment by warmachine028** [comment]
> @kumarankm Is the mentor of this issue

Hey! Can you ask him to have a look into my PR?

**Comment by warmachine028** [comment]
> > > > > @kumarankm Is the mentor of this issue
> > > > 
> > > > 
> > > > Hey! Can you ask him to have a look into my PR?
> > > 
> > > 
> > > Tag your mentor in your PR
> > 
> > 
> > I've tagged him in my PR as you asked me to .. but he is not replying me
> 
> Proceed with other issues for now. Your PR will be reviewed by mentor shortly

No... Actually he mentioned some issue with my PR .. I'll resolve it and make it again by today ...
